csrc/quantization/machete/machete_pytorch.cu

```diff
@@ -0,0 +1,84 @@
+#include "machete_mm_launcher.cuh"
+#include "machete_prepack_launcher.cuh"
+#include "core/scalar_type.hpp"
+
+namespace machete {
+
+using namespace vllm;
+
+//
+//  Utils (type dispatching)
+//
+
+template <typename Fn>
+static auto scalar_type_dispatch(ScalarType const& type, Fn fn) {
```

Comment:
> But I thought only kU4 and kS4 were supported?

we generate kernels for kU4, kU8, kU4B8, and kU8B128 in `generate.py`

Im generally not a fan of creating utilities until they would used in multiple spots, Im happy to create a utility once sombody else (or I do something similar) in a follow up PR. 
Also a utilty like this is a bit challenging since currently `vllm::kU4`, `vllm::kU8` etc. can't be used as template parameter in C++17 until https://github.com/vllm-project/vllm/pull/7323 lands, after that this becomes alot easier.

Reply:
> only kU4 and kS4 were supported?
oh I see what you are saying! `supported_types` is out of sync, fix

Reply:
Okay that's very reasonable re:utility, cool